fondant/dataset.py

```diff
@@ -74,26 +64,19 @@ def _load_index(self):
         index = self.manifest.index
         # get remote path
         remote_path = index.location
-
-        df = dd.read_parquet(remote_path)
-
-        if list(df.columns) != ["id", "source"]:
-            raise ValueError(
-                f"Index columns should be 'id' and 'source', found {df.columns}"
-            )
+        # load index from parquet, expecting id and source columns
+        df = dd.read_parquet(remote_path, columns=["id", "source"])
 
         return df
 
     def load_data(self, spec: FondantComponentSpec) -> dd.DataFrame:
-        subset_dfs = []
+        # load index into dataframe
+        df = self._load_index()
         for name, subset in spec.input_subsets.items():
             fields = list(subset.fields.keys())
             subset_df = self._load_subset(name, fields)
-            subset_dfs.append(subset_df)
-
-        # return a single dataframe with column_names called subset_field
-        # TODO perhaps leverage dd.merge here instead
-        df = dd.concat(subset_dfs)
+            # left joins -> filter on index
+            df = dd.merge(df, subset_df, how="left")
```

Comment:
does it automatically know which columns names to join on?

Reply:
might be more efficient if we use the `_set_index()` method on both datasets before joining: 
https://docs.dask.org/en/stable/dataframe-best-practices.html

Reply:
The `set_index()` is persisted when writing and then reading to/from parquet. So Ideally this is already done upstream when the subset is created.

Reply:
> does it automatically know which columns names to join on?

It joins on columns with the same name, an optimisation is definitely to join on the index.

Reply:
Yeah we need to make sure that all subsets are always aligned. Since we might not be reading all subsets here, we should do this when originally writing the subsets, all the way back to the loading component, which should set the index for the different subsets it generates.

Reply:
Is there a reason to not define here that it should join on the index?